packages/connect/src/impl/core-in-module.ts

```diff
@@ -0,0 +1,298 @@
+import EventEmitter from 'events';
+
+// NOTE: @trezor/connect part is intentionally not imported from the index due to NormalReplacementPlugin
+// in packages/suite-build/configs/web.webpack.config.ts
+import * as ERRORS from '@trezor/connect/src/constants/errors';
+import {
+    POPUP,
+    IFRAME,
+    UI,
+    UI_EVENT,
+    DEVICE_EVENT,
+    RESPONSE_EVENT,
+    TRANSPORT_EVENT,
+    BLOCKCHAIN_EVENT,
+    createErrorMessage,
+    UiResponseEvent,
+    CoreEventMessage,
+    CallMethodPayload,
+    CORE_EVENT,
+} from '../events';
+import type { ConnectSettings, ConnectSettingsPublic, DeviceIdentity, Manifest } from '../types';
+import { ConnectFactoryDependencies, factory } from '../factory';
+import { Log, initLog } from '../utils/debug';
+import { DeferredManager, createDeferredManager } from '@trezor/utils';
+
+import { parseConnectSettings } from '../data/connectSettings';
+
+export class CoreInModule implements ConnectFactoryDependencies<ConnectSettingsPublic> {
+    public eventEmitter = new EventEmitter();
+    protected _settings: ConnectSettings;
+
+    private _log: Log;
+    private _coreManager?: any;
+    private _messagePromises: DeferredManager<{
+        id: number;
+        success: boolean;
+        payload: any;
+        device?: DeviceIdentity;
+    }>;
+
+    private readonly boundOnCoreEvent = this.onCoreEvent.bind(this);
+
+    public constructor() {
+        this._settings = parseConnectSettings();
+        this._log = initLog('@trezor/connect-web');
+        this._messagePromises = createDeferredManager({ initialId: 1 });
+    }
+
+    private async initCoreManager() {
+        const connectSrc = this._settings.connectSrc;
+        const { initCoreState, initTransport } = await import(
+            /* webpackIgnore: true */ `${connectSrc}js/core.js`
+        ).catch(_err => {
+            this._log.error('_err', _err);
```

Comment:
what is the reason for dynamic load here?

Reply:
Making it flexible, so in the context of `suite-web` connectSrc is going to be different as if we use it somewhere else. 
How would you do it?